contracts/LoanCore.sol

```diff
@@ -258,32 +276,52 @@ contract LoanCore is
      * @param payer                 The party repaying the loan.
      * @param _amountFromPayer      The amount of tokens to be collected from the repayer.
      * @param _amountToLender       The amount of tokens to be distributed to the lender (net after fees).
+     * @param _interestAmount       The interest amount to be paid.
+     * @param _paymentToPrincipal   The portion of the repayment amount that goes to principal.
      */
     function forceRepay(
         uint256 loanId,
         address payer,
         uint256 _amountFromPayer,
-        uint256 _amountToLender
+        uint256 _amountToLender,
+        uint256 _interestAmount,
+        uint256 _paymentToPrincipal
     ) external override onlyRole(REPAYER_ROLE) nonReentrant {
-        LoanLibrary.LoanData memory data = _handleRepay(loanId, _amountFromPayer, _amountToLender);
-
-        // Do not send collected principal, but make it available for withdrawal by a holder of the lender note
-        noteReceipts[loanId] = NoteReceipt({
-            token: data.terms.payableCurrency,
-            amount: _amountToLender
-        });
+        (LoanLibrary.LoanData memory data, bool loanRepaid) = _handleRepay(
+            loanId,
+            _amountFromPayer,
+            _amountToLender,
+            _interestAmount,
+            _paymentToPrincipal
+        );
+
+        // DO NOT send collected principal, but make it available for withdrawal by a holder of the LenderNote
+        NoteReceipt storage receipt = noteReceipts[loanId];
+        if (receipt.token == address(0)) {
+            receipt.token = data.terms.payableCurrency;
+            receipt.amount = _amountToLender;
+        } else {
+            receipt.amount += _amountToLender;
+        }
 
-        // Get promissory notes from two parties involved, then burn
-        // borrower note _only_ - do not burn lender note until receipt
-        // is redeemed
+        // get owner of the BorrowerNote, and burn only if loan is fully repaid.
+        // DO NOT burn LenderNote until receipt is redeemed
         address borrower = borrowerNote.ownerOf(loanId);
-        borrowerNote.burn(loanId);
+        if (loanRepaid) borrowerNote.burn(loanId);
 
-        // Collect from borrower and redistribute collateral
+        // collect repayment amount from payer
         _collectIfNonzero(IERC20(data.terms.payableCurrency), payer, _amountFromPayer);
-        IERC721(data.terms.collateralAddress).safeTransferFrom(address(this), borrower, data.terms.collateralId);
 
-        emit LoanRepaid(loanId);
+        if (loanRepaid) {
+            // if loan is completely repaid redistribute collateral and emit repaid event
+            IERC721(data.terms.collateralAddress).safeTransferFrom(address(this), borrower, data.terms.collateralId);
+
+            emit LoanRepaid(loanId);
+        } else {
+            // if loan is not completely repaid emit payment event
+            emit LoanPayment(loanId);
```

Comment:
I would emit `LoanPayment` in both scenarios (so on closure, we emit 2 events in this fn)